src/utils.js

```diff
@@ -70,24 +71,31 @@ const generateHTML = (fileData, filePath, outputFolder) => {
     </body>
     </html>`;
 
-    // Write text to HTML file -- Need to update for custom output path
+    // Write input to HTML file -- Need to update for custom output path
     try {
-      fs.writeFileSync(`${outputFolder}/${path.basename(filePath[i], ".txt")}.html`, html);
+      const fileType = filePath[i].split(".")[1];
```

Comment:
File type is stored in index = 2:  `filePath[i].split(".")[2]`

Reply:
Unless I'm mistaken, I don't think there is an index of value 2, since `split(".")` would return an array of only two elements (index 0 and 1), not three.

If running `node src/index.js examples`, the value of `filePath` at runtime is:
```ts
[
  'examples/example1.txt',
  'examples/example2.txt',
  'examples/markdown_Sample.md',
  'examples/til_Sample.txt'
]
```

`filePath[0]` would return `'examples/example1.txt'`

`filePath[0].split(".")` would return:
```ts
[
  'examples/example1',
  'txt'
]
```

`filePath[0].split(".")[1]` would return the filetype `'txt'` but `filePath[0].split(".")[2]` would return `undefined` since there's no 3rd element in this array.

If running `node src/index.js examples` with your proposed change for example, the program incorrectly generates the following files:
- example1.txt.html
- example2.txt.html
- markdown_Sample.md.html
- til_Sample.txt.html

Compared to running it with the PR's code:
- example1.html
- example2.html
- markdown_Sample.html
- til_Sample.html

This would break step 9. of the [release 0.1 spec](https://github.com/Seneca-CDOT/topics-in-open-source-2023/wiki/release-0.1):
> 9. your tool should generate one .html output file for each input file. For example, if I run the tool on doc.txt a new doc.html file will be generated.

Reply:
I see. The issue arises based on how the user inputs the path. I'm used to entering the current folder with the path: `./examples/markdown_Sample.md`. In this situation, the `.split(".")` command would be broken up into 3 parts and the extension would be within index = 2,  and that was what I was initially seeing.

I would suggest using a method from the path module to only grab the extension of the file regardless if the path started with `./` or not.

Something like this:
```
File: utils.js
75:     try {
76:       // const fileType = filePath[i].split(".")[1];   <--   Can delete
77:       
78:       fs.writeFileSync(`${outputFolder}/${path.basename(filePath[i], path.extname(filePath[i]))}.html`, html);
79:       // need to update for custom output path
80:       console.log(`File successfully written at: ${outputFolder}/${path.basename(filePath[i], path.extname(filePath[i]))}.html`);
81:     } catch (err) {

```

Reply:
Ah good catch, my solution isn't robust enough to handle that. That looks like a better solution, will implement.

Reply:
Replaced `split` with `path.extname` to extract the extension name and ready for review.

Reply:
Looks good!